### PR TITLE
usm: http: Flushing last http transaction if we got a new request

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Flushes previous http state, if we got a new request `packet_type == HTTP_REQUEST` and we previously we saw a request / response.

### Motivation

In case we're seeing first a response and then a request, we'll match them together and have "off by one"

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/ebpf/c/protocols/http/http.h [hunk 1]
```diff
@@ -109,7 +109,7 @@ static __always_inline http_transaction_t *http_fetch_state(http_transaction_t *
 }
 
 static __always_inline bool http_should_flush_previous_state(http_transaction_t *http, http_packet_t packet_type) {
-    return (packet_type == HTTP_REQUEST && http->request_started) ||
+    return (packet_type == HTTP_REQUEST && (http->request_started || http->response_status_code)) ||
```
Comment (val06) [inline pkg/network/ebpf/c/protocols/http/http.h:112]: please add a comment that explains in words the state machine you have here, meaning what situations could occur in the real world traffic and how should we handle that. (preferably split each potential "state" in a separate bullet / line

**Review by val06 — CHANGES_REQUESTED** [review]
cred

**Review by val06 — APPROVED** [review]

**Comment by p-lambert** [comment]
@guyarb isn't the scenario described in the PR exactly the one we're testing here?
https://github.com/DataDog/datadog-agent/blob/f735cb4e205385b03bc1f3befd42db694afe2baf/pkg/network/usm/monitor_test.go#L516
If it's not can you please add a regression test following the pattern above?

**Comment by guyarb** [comment]
> @guyarb isn't the scenario described in the PR exactly the one we're testing here?
> 
> https://github.com/DataDog/datadog-agent/blob/f735cb4e205385b03bc1f3befd42db694afe2baf/pkg/network/usm/monitor_test.go#L516
> 
> 
> If it's not can you please add a regression test following the pattern above?

The original code works as we're overriding original response by the new response
TBH, testing is not feasible, as it involves processing the incomplete buffers, which are not exported in the context of the monitor

**Pull request merged**